article/_posts/2020-06-04-parallelization.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: review
+title: "Demystifying Parallel and Distributed Deep Learning"
+tags: deep-learning
+author: "Carl Lemaire"
+cite:
+    authors: "TAL BEN-NUN, TORSTEN HOEFLER"
+    title:   "Demystifying Parallel and Distributed Deep Learning: An In-Depth Concurrency Analysis"
+    venue:   "ACM Computing Surveys"
+pdf: "https://arxiv.org/pdf/1802.09941.pdf"
+---
+
+In this survey, the different forms of parallelism and concurrency involved in deep learning are described. The authors cover operator-level parallelism, network-level parallelism, and parallelism in model selection. In this review, I will focus on the network-level parallelism.
+
+# Types of network-level parallelism
+
+The authors describe 3 types of parallelism: data parallelism, model parallelism and pipelining. Calling pipelining a form of parallelism is debatable, since of its inherently sequential nature.
+
+**It's important to note that the term "model parallelism" is often used when in fact "layer pipelining" should be used.** While in both cases, the network is partitioned across multiple devices, the two concepts are different. The first is parallel, the other is not.
+
+![](/article/images/parallelization/fig14.jpg)
+
+## Data parallelism
+
+This form of parallelism is very common because of its simplicity. It consists in replicating the network on all devices, and distributing the batch equally to feed the replicas. It is optimal when the ratio of activations size to weight size is high (i.e. the weights are relatively small).
+
+## Model parallelism
+
+Some claim that model parallelism consists in partitioning the network across multiple devices; this is incorrect. Model parallelism consists in executing multiple **parallel branches** of a network on multiple devices. Not all network or layer types can benefit from this form of parallelism: notably, a typical CNN architecture are considered incompatible.
```

Comment:
_(...) notably, a typical CNN architecture are considered incompatible._ -> _(...) notably, a typical CNN architecture is considered incompatible._